src/core/index.ts

```diff
@@ -356,13 +356,16 @@ export class Core {
     }): AssetScheme {
         const {
             shardId,
-            metadata,
             supply,
             approver = null,
             registrar = null,
             allowedScriptHashes = null,
             pool = []
         } = params;
+        const metadata =
+            typeof params.metadata === "string"
+                ? params.metadata
+                : JSON.stringify(params.metadata);
```

Comment:
`metadata` can be a non-object here. I think you should change `checkMetadata` and call it before this.